### DISCUSSION TITLE
The metadata is returned as empty.

### DISCUSSION THREAD

**Discussion by shige0309** [supabase/supabase#16815] [Questions] [answered]
What I want to do is set up functions so that when a user registers, I want to register the user_name and tel in the profile table. I am currently trying out different methods to achieve this.

//functions
begin
insert into public.profile(id)
values(new.id);
return new;
end;

I want to retrieve the metadata, so I implemented the solution based on the URL below. However, the metadata is returned as empty.

https://supabase.com/docs/guides/auth/managing-user-data#accessing-user-metadata

I checked various sites and tried different methods, but none of them worked. I would appreciate it if you could instruct me on how to register the user_name and tel in the profile table when registering a user.

I look forward to your guidance.

・Here's the actual code. I've excerpted parts of it.
`import { useMutation } from "react-query";
import { createClientComponentClient } from "@supabase/auth-helpers-nextjs";
import { Database } from "../types/supabase";

type Auth = {
  email: string;
  password: string;
};

export const useMutateAuth = () => {
  const supabase = createClientComponentClient<Database>();
  const loginMutate = useMutation(
    async ({ email, password }: Auth) => {
      const { data, error } = await supabase.auth.signUp({
        email,
        password,
        options: {
          data: {
            user_name: "John",
            tel: "090-1234-5678",
          },
        },
      });
      if (error) throw new Error(error.message);
      console.log("data?.user?.user_metadata", data?.user?.user_metadata);
    },
    {
      onError: (error: Error) => {
        alert(
          error.message
        );
      },
    }
  );
}`

**Comment by GaryAustin1** [comment]
Did you add any code to store the metadata?
If you goto the SQL editor in the dashboard under Quickstarts there is a user management starter.  That shows a function that handles the user metadata.
```
begin
  insert into public.profiles (id, full_name, avatar_url)
  values (new.id, new.raw_user_meta_data->>'full_name', new.raw_user_meta_data->>'avatar_url');
  return new;
end;
```
Edit: You need to change the name of the metadata to match your json you are passing in and have columns in your table to store the data in.

**Reply by shige0309** [reply]
Mr. GaryAustin1,

Thank you for your guidance.
I've written the code as you suggested.

However, both first_name and tel end up being null.
The creation of a new user proceeds without any issues.

I've checked various things, but everything seems to be in order.
I'm completely puzzled as to why they become null.

I would greatly appreciate any further insights you can provide.

Thank you in advance.

`

/=== SQL ====/
-- inserts a row into public.profiles
create function public.create_profile_for_user()
returns trigger
language plpgsql
security definer set search_path = public
as $$
begin
  insert into public.profiles (id, first_name, tel)
  values (new.id,
  new.raw_user_meta_data->>'first_name',
  new.raw_user_meta_data->>'tel');
  return new;
end;
$$;

-- trigger the function every time a user is created
create trigger create_profile_for_user_trigger
  after insert on auth.users
  for each row execute procedure public.create_profile_for_user();



// ==== useMutation.ts Excerpt ====//
import { useMutation } from "react-query";
import { createClientComponentClient } from "@supabase/auth-helpers-nextjs";
import { Database } from "../types/supabase";

type Auth = {
  email: string;
  password: string;
};

export const useMutateAuth = () => {
  const supabase = createClientComponentClient<Database>();
  const loginMutate = useMutation(
    async ({ email, password }: Auth) => {
      const { data, error } = await supabase.auth.signUp({
        email,
        password,
        options: {
          data: {
            first_name: "John",
            tel: "123456789"
          },
        },
      });
      if (error) throw new Error(error.message);
    },
    {
      onError: (error: Error) => {
        alert(
          error.message
        );
      },
    }
  );`

**Reply by GaryAustin1** [reply, marked as answer]
I don't see anything obviously wrong.
You can add `raise log 'new = %',new;` after begin and then look in the dashboard Postgres logs to see if new contains the values to narrow down possibly.

**Reply by shige0309** [reply]
Hello GaryAustin1,

Thank you for your guidance.

Upon further inspection, I realized I had mistakenly modified the sign-in section. Once I corrected the sign-up section, I was able to register successfully.

I greatly appreciate all the assistance you've provided.
Thank you.
